TESTS/mbed_drivers/rtc/main.cpp

```diff
@@ -16,41 +16,423 @@
-        wait(1);
-    }
+    return rtc_time_val;
 }
 
-Case cases[] = {
-    Case("RTC strftime", test_case_rtc_strftime),
-};
+/* Stub of RTC write function. */
+static void write_rtc_stub(time_t t)
+{
+   rtc_write_called = true;
+
+   rtc_time_val = t;
+}
+
+/* Stub of RTC init function. */
+static void init_rtc_stub(void)
+{
+    rtc_init_called = true;
+}
+
+/* Stub of RTC isenabled function. */
+static int isenabled_rtc_stub(void)
+{
+    rtc_isenabled_called = true;
+
+    return rtc_enabled_ret;
+}
+
+/* This test verifies if attach_rtc provides availability to
+ * connect specific RTC driver functions.
+ *
+ * This is unit test to verify if correct functions are used
+ * to support RTC.
+ *
+ * Given specific RTC driver functions have been attached (stubs).
+ * When set_time/time functions are called.
+ * Then set_time/time functions use attached RTC functions.
+ */
+void test_attach_RTC_funtions()
+{
+    time_t seconds = 0;
+
+    /* Attache RTC read/write/init/isenabled stubs. */
+    attach_rtc(read_rtc_stub, write_rtc_stub, init_rtc_stub, isenabled_rtc_stub);
+
+    /* Init stub variables/set to unexpected. */
+    rtc_write_called = false;
+    rtc_init_called = false;
+
+    /* Call set_time() function. We expect that init and write RTC stubs
+     * will be executed.
+     */
+    set_time(CUSTOM_TIME_1);
+
+    /* Verify results. */
+    TEST_ASSERT_EQUAL(true, rtc_write_called);
+    TEST_ASSERT_EQUAL(true, rtc_init_called);
+
+    /* Init stub variables/set to unexpected. */
+    rtc_time_val = CUSTOM_TIME_1;
+    rtc_enabled_ret = true;
+    rtc_isenabled_called = false;
+    rtc_read_called = false;
+
+    /* Call time() function. We expect that isenabled and read RTC stubs
+     * are be executed.
+     */
+    time(NULL);
+
+    /* Verify results. */
+    TEST_ASSERT_EQUAL(true, rtc_isenabled_called);
+    TEST_ASSERT_EQUAL(true, rtc_read_called);
+
+    /* This part of the test can be executed only on RTC devices. */
+
+    /* Restore env. */
+    attach_rtc(rtc_read, rtc_write, rtc_init, rtc_isenabled);
+
+    /* Set to unexpected. */
+    rtc_write_called = false;
+    rtc_init_called = false;
+    rtc_isenabled_called = false;
+    rtc_init_called = false;
+
+    /* Set time. */
+    set_time(CUSTOM_TIME_1);
+
+    /* Get time. */
+    seconds = time(NULL);
+
+    /* Stub RTC functions should not be called now. */
+    TEST_ASSERT_EQUAL(false, rtc_isenabled_called);
+    TEST_ASSERT_EQUAL(false, rtc_init_called);
+    TEST_ASSERT_EQUAL(false, rtc_write_called);
+    TEST_ASSERT_EQUAL(false, rtc_init_called);
+
+    /* Check if time has been successfully set and retrieved. */
+    TEST_ASSERT_EQUAL(CUSTOM_TIME_1, seconds);
+}
+
+/* This test verifies if time() function returns
+ * current time when all RTC functions are
+ * defined and RTC is enabled.
+ *
+ * Note: Stubs are used instead of original RTC functions.
+ *
+ * Given environment has RTC functions
+ * defined and RTC is enabled.
```

Comment:
nit: doesn't need to be broken into two lines, we have limit of 120chars https://docs.mbed.com/docs/mbed-os-handbook/en/latest/cont/code_style/

Reply:
Description of the test cases have been corrected.